protoc-gen-swagger/genswagger/template.go

```diff
@@ -1448,6 +1456,109 @@ func processExtensions(inputExts map[string]*structpb.Value) ([]extension, error
 	return exts, nil
 }
 
+func validateHeaderType(headerType string) error {
+	// The type of the object. The value MUST be one of "string", "number", "integer", "boolean", or "array"
+	// See: https://github.com/OAI/OpenAPI-Specification/blob/3.0.0/versions/2.0.md#headerObject
+	// Note: currently not implementing array as we are only implementing this in the operation response context
+	switch headerType {
+	case
+		"string",
+		"number",
+		"integer",
+		"boolean":
+		return nil
+	}
+	return fmt.Errorf("the provided header type %q is not supported", headerType)
+}
+
+func validateDefaultValueType(headerType string, defaultValue string) error {
+	switch headerType {
+	case
+		"string":
```

Comment:
Nit; this are generally put on the same line in Go, not sure why gofmt didn't change this.
```suggestion
	case "string":
```

Reply:
Understood.... is this true when you have a list of statements for an argument?

i.e 
```
	switch s {
	case
		"nan",
		"inf",
		"infinity":
		return fmt.Errorf("the provided number %q is not a valid JSON number", s)
	}
```
or
```
	switch s {
	case "nan", "inf", "infinity":
		return fmt.Errorf("the provided number %q is not a valid JSON number", s)
	}
```

Reply:
Generally, still the latter, but we won't need it here anyway :).